### ISSUE TITLE
Question: Engineering Presets?

### ISSUE THREAD

**Issue by WumbleMinky** [daid/EmptyEpsilon#452]
I've been playing around with the code and added some preset buttons on the Engineering Screen. All they do is save all of the systems power and coolant levels and then set them when the button is pressed.

My question is: is this a function that you want as part of the design? If so, I can fork the code, make the changes and submit a pull request, but I don't want to bother if it is not something that is wanted. I know it's something that my friend really likes in Artemis, which is why I tried setting it up in EE.

Here is a link to a screenshot [http://imgur.com/a/2hjo1](url).

**Comment by nallath** [comment]
I'm not really a fan of this feature, as it might make it a bit too simple for engineering. On the other hand, it would make it a lot easier for noobies to get started. It might be a good idea to make it something that can be toggled (like shield frequencies)

**Comment by kwadroke** [comment]
+1 for the toggle for this.
Would be good for battle, cruising, and low power mode.
Course with the custom buttons, this could be added to the scripts, except maybe for setting the values.
Speaking of custom buttons: these buttons might need to be moved somewhere else as they might overlap with the custom buttons.

**Comment by WumbleMinky** [comment]
I don't think it would make engineering simple (especially since there are only 4 presets), but the toggle option is still a good idea. The idea just seems more realistic to me: ships **that** advanced would have some form of saved configurations.

I didn't realize about the custom buttons (I didn't even know that was an option). I'll see about moving the preset buttons around.

**Comment by nallath** [comment]
Realistic? Maybe. I'm just concerned it makes engineering too easy. Right now the speed by which engineering can switch configuration is limited by the skills of the engineer. If it's as simple as pressing a button, it removes that entirely.

**Comment by CaptLapadite** [comment]
A lot of our players were looking for that king of feature. But since engineering relies mostly on fine tuning and sparing energy, IMO I do not have the impression that this new feature would break the engineering gameplay. I think it would mostly be used to go back to normal settings on green conditions or typical fights (beam, missile, jump). Having just 4 presets is also quite limited, and I think its good that way. An option to toggle it is great tho. 

Maybe adding a delay to go from the actual setting to the preset would be great ? for exemple instead of changing from 0 to 200% in jump drive by a simple click of your "emergency jump" preset, it could take a few seconds to go from one value to the other.

When playing in a solo ship, my friends and I usually prefer the "3 players ship" template for more options on the ship, always switching roles during the game. I would gladly use theses presets in engineering for that regard.

**Comment by kwadroke** [comment]
For me, this functionality would be good for new players at conventions.
For regular game play with more experienced crew, I'd disable it.

I'm working on creating simplified preset buttons in a LUA script where I can add it as a require to the scenarios as needed. I just need to test it out to see if it works.

**Comment by Wrongtown** [comment]
+1 for having this added as a "Enable if you like it, don't enable if you don't" feature. 

Having a quick toggle to get to a known state doesn't mean the engineer isn't then going to be further tweaking from that baseline if they want to really keep the ship running efficiently.

**Comment by WumbleMinky** [comment]
I like the feedback. Thanks.

I've made a few updates and moved the buttons around. I've also added a toggle feature on the server startup screen. Have a look and let me know what everyone thinks.

[http://imgur.com/a/ouQ1v](http://imgur.com/a/ouQ1v)

**Comment by CaptLapadite** [comment]
well done. 

I like how the "save" button is off with the alignment with the different engineering modules. 
I would do the same for the preset numbers just to make sure it is not wrongly associated with the modules.

**Comment by WumbleMinky** [comment]
That's a good point. Consider it done. :)

**Comment by CousinsP** [comment]
Did this get included in the game at all? How can I get it?
Thanks.

**Comment by oznogon** [comment]
Only seems to be on @WumbleMinky's fork, no pull request for it here that I can find and no updates on the fork since 2017: https://github.com/daid/EmptyEpsilon/compare/master...WumbleMinky:engineering-presets

**Comment by WumbleMinky** [comment]
Yeah it fell off my plate and I kind of forgot about it. I'm not even sure it would work with newer versions of the code. Plus, I'm not very familiar with git (or repos in general) so I'm not sure how pull requests work. If there is any interest, I can look into it.

**Comment by daid** [comment]
I've always been against engineering presets, I think they take away instead of add to the game.
